docs/tutorials/podman_pasta_ipv6_with_snid_ipv4.md

```diff
@@ -0,0 +1,761 @@
+podmanhost01        	  IN	    AAAA	  2001:db8:0000:0001:0000:0000:0000:0100
+
+whoami			            IN	    A       198.51.100.10
+whoami			            IN      AAAA    2001:db8:0000:0001:0000:0000:0001:0001
+
+```
+
+Add to `/etc/systemd/resolved.conf` on the DNS Server Host:
+```
+[Resolve]
+# Use local BIND Name Server, which will forward to Hetzner Recursive Name Servers for external (non-managed) Domains
+DNS=127.0.0.1
+Domains=MYDOMAIN.TLD
+DNSSEC=yes
+DNSOverTLS=opportunistic
+
+# Set to NO since there is already a DNS Server on the System (BIND/NAMED)
+DNSStubListener=no
+```
+
+Add to `/etc/systemd/resolved.conf` on the Podman Host:
+```
+[Resolve]
+DNS=172.16.1.3 2001:db8:0000:0001:0000:0000:0000:0003
+#FallbackDNS=
+Domains=MYDOMAIN.TLD
+DNSSEC=yes
+DNSOverTLS=opportunistic
+```
+
+On both Hosts, run:
+```
+systemctl restart systemd-resolved
+```
+
+In this way, the correct IPv4 <-> IPv6 Translation should be performed and the correct Application should now be able to be reacheable.
+
+# Compose File Setup
+Two different Approaches for the Compose File are Possible and both Work.
+
+## env File
+In order to simplify the Setup, it is Proposed to store the IP Address in an `.env` File.
+
+```
+echo 'APPLICATION_IPV6_ADDRESS="2001:db8:0000:0001:0000:0000:0001:0001"' >> .env
+```
+
+In this way, the `compose.yml` can be evaluated with the `${APPLICATION_IPV6_ADDRESS}` Variable replaced by its Value, by running:
+```
+podman-compose config
+```
+
+## Compose with Port Mapping + Minimal Pasta Line
+With this Method, `podman ps` will show the Open Ports in its Output.
+
+```
+services:
+  whoami-caddy:
+    image: caddy:latest
+    #image: lucaslorentz/caddy-docker-proxy:2.9-alpine
+    pull_policy: "missing"
+    container_name: whoami-caddy
+    restart: "unless-stopped"
+    security_opt:
+      - no-new-privileges:true
+      - label=type:container_runtime_t
+    ports:
+      - target: 80
+        host_ip: "[${APPLICATION_IPV6_ADDRESS}]"
+        published: 80
+        protocol: tcp
+      - target: 443
+        host_ip: "[${APPLICATION_IPV6_ADDRESS}]"
+        published: 443
+        protocol: tcp
+      - target: 443
+        host_ip: "[${APPLICATION_IPV6_ADDRESS}]"
+        published: 443
+        protocol: udp
+    network_mode: "pasta:--ipv6-only"
+    volumes:
+ #     - /run/user/1001/podman/podman.sock:/var/run/docker.sock:rw,z
+      - ./Caddyfile:/etc/caddy/Caddyfile:ro,z
+      - ~/containers/data/whoami-caddy:/data:rw,z
+      - ~/containers/log/whoami-caddy:/var/log:rw,z
+      - ~/containers/config/whoami-caddy:/config:rw,z
+      - ~/containers/certificates/letsencrypt:/certificates:ro,z
+    environment:
+      - CADDY_DOCKER_CADDYFILE_PATH=/etc/caddy/Caddyfile
+
+  # Proxy to container
+  whoami-application:
+    image: traefik/whoami
+    pull_policy: "missing"
+    container_name: whoami-application
+    restart: "unless-stopped"
+    network_mode: "service:whoami-caddy"
+    environment:
+      - WHOAMI_PORT_NUMBER=8080
```

Comment:
cleaned up:
```yaml
---
services:
  whoami-caddy:
    image: caddy:latest
    command: caddy reverse-proxy --from ${HOSTNAME} --to 'http://[::1]:8080'
    network_mode: "pasta:--ipv6-only"
    ports:
      - target: 80
        published: 80
        host_ip: "[${APPLICATION_IPV6_ADDRESS}]"
        protocol: tcp
      - target: 443
        published: 443
        host_ip: "[${APPLICATION_IPV6_ADDRESS}]"
        protocol: tcp
      - target: 443
        published: 443
        host_ip: "[${APPLICATION_IPV6_ADDRESS}]"
        protocol: udp
    volumes:
      - caddy-data:/data

  whoami-application:
    image: traefik/whoami
    network_mode: "service:whoami-caddy"
    environment:
      - WHOAMI_PORT_NUMBER=8080

volumes:
  caddy-data:
```
or even shorter:
```yaml
---
services:
  whoami-caddy:
    image: caddy:latest
    command: caddy reverse-proxy --from ${HOSTNAME} --to 'http://[::1]:8080'
    network_mode: "pasta:--ipv6-only"
    ports:
      - "[${APPLICATION_IPV6_ADDRESS}]:80:80/tcp"
      - "[${APPLICATION_IPV6_ADDRESS}]:443:443/tcp"
      - "[${APPLICATION_IPV6_ADDRESS}]:443:443/udp"
    volumes:
      - caddy-data:/data

  whoami-application:
    image: traefik/whoami
    network_mode: "service:whoami-caddy"
    environment:
      - WHOAMI_PORT_NUMBER=8080

volumes:
  caddy-data:
```

Reply:
I split both the compose part and the Caddy Configuration into two Subsections:
1. Using Default Caddy Configuration
2. Using Customized Caddy Configuration

I implemented your short(est) version in the first subsection (which probably fits most users), while keeping the one I currently have (and use on my Server) as second.